tagstudio/src/core/media_types.py

```diff
@@ -0,0 +1,495 @@
+# Copyright (C) 2024 Travis Abendshien (CyanVoxel).
+# Licensed under the GPL-3.0 License.
+# Created for TagStudio: https://github.com/CyanVoxel/TagStudio
+
+import logging
+import mimetypes
+from enum import Enum
+from pathlib import Path
+
+logging.basicConfig(format="%(message)s", level=logging.INFO)
+
+
+class MediaType(str, Enum):
+    """Names of media types."""
+
+    ADOBE_PHOTOSHOP: str = "adobe_photoshop"
+    AFFINITY_PHOTO: str = "affinity_photo"
+    ARCHIVE: str = "archive"
+    AUDIO_MIDI: str = "audio_midi"
+    AUDIO: str = "audio"
+    BLENDER: str = "blender"
+    DATABASE: str = "database"
+    DISK_IMAGE: str = "disk_image"
+    DOCUMENT: str = "document"
+    FONT: str = "font"
+    IMAGE_ANIMATED: str = "image_animated"
+    IMAGE_RAW: str = "image_raw"
+    IMAGE_VECTOR: str = "image_vector"
+    IMAGE: str = "image"
+    INSTALLER: str = "installer"
+    MATERIAL: str = "material"
+    MODEL: str = "model"
+    PACKAGE: str = "package"
+    PDF: str = "pdf"
+    PLAINTEXT: str = "plaintext"
+    PRESENTATION: str = "presentation"
+    PROGRAM: str = "program"
+    SHORTCUT: str = "shortcut"
+    SOURCE_ENGINE: str = "source_engine"
+    SPREADSHEET: str = "spreadsheet"
+    TEXT: str = "text"
+    VIDEO: str = "video"
+
+
+class MediaCategory:
+    """An object representing a category of media.
+
+    Includes a MediaType identifier, extensions set, and IANA status flag.
+
+    Args:
+        media_type (MediaType): The MediaType Enum representing this category.
+
+        extensions (set[str]): The set of file extensions associated with this category.
+            Includes leading ".", all lowercase, and does not need to be unique to this category.
+
+        is_iana (bool): Represents whether or not this is an IANA registered category.
+    """
+
+    def __init__(
+        self,
+        media_type: MediaType,
+        extensions: set[str],
+        is_iana: bool = False,
+    ) -> None:
+        self.media_type: MediaType = media_type
+        self.extensions: set[str] = extensions
+        self.is_iana: bool = is_iana
```

Comment:
Since this is just a data wrapper, I'd suggest to use `dataclass` instead. It will save you from writing the constructor and assigning the values in it manually.

```python
@dataclass(frozen=True)
class MediaCategory:
    media_type: MediaType
    extensions: set[str]
    is_iana: bool # maybe assign default value, so you dont have to initialize it explicitly half of the times?
```

example of usage eg. here:

https://github.com/TagStudioDev/TagStudio/blob/49d071cf2b2955c3a5dfb96e7df0661d9cac0b44/tagstudio/src/qt/modals/folders_to_tags.py#L33-L37